microcosm_metrics/factories.py

```diff
@@ -0,0 +1,67 @@
+"""
+Factories for statsd/metrics clients.
+
+"""
+from datadog import DogStatsd
+from microcosm.api import defaults
+from statsd import StatsClient
+
+
+def attach(graph, statsd):
+    """
+    Attach statsd client to `metrics` key for pseudo polymorphism.
+
+    """
+    graph.metrics = statsd
```

Comment:
I get why this is a good idea considering we've switched 3rd parties before and inevitably will again, but I do feel like there's something we'll want to enhance here.... I'll think about it.